client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -1,20 +1,46 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import {
   DatePicker,
   DatePickerProps,
 } from '@mui/x-date-pickers/DatePicker/DatePicker';
 import { BasicTextInput } from '../../TextInput/BasicTextInput';
 import { StandardTextFieldProps, TextFieldProps } from '@mui/material';
 import { useAppTheme } from '@common/styles';
+import { DateUtils } from '@common/intl';
+import { useDebounceCallback } from '@common/hooks';
 
 export const BaseDatePickerInput: FC<
-  Omit<DatePickerProps<Date>, 'renderInput'>
-> = props => {
+  Omit<DatePickerProps<Date>, 'renderInput' | 'value'> & {
+    onChange(date: Date): void;
+    value: Date | string | null;
+  }
+> = ({ disabled, onChange, value, ...props }) => {
   const theme = useAppTheme();
+  const [internalValue, setInternalValue] = useState<Date | null>(null);
+
+  useEffect(() => {
+    // This sets the internal state from parent when first loading (i.e. when
+    // the internal date is still empty)
```

Comment:
Nit: Not only when first loading but potentially when value changes, right? not sure if this is important for the logic though?

Reply:
No, it's only on first load, hence the `internalValue === null` check. Otherwise the internalValue gets set twice on userInput -- once due to local `onChange` and again after parent value changes (after debounce).

If we need it to respond to external changes in the parent value, we can address that once we understand the specific case, but for now, it's always set from user input except on first load, where it loads the saved value.